File: src/main/java/io/pageobject/generator/name/LinkPostfixNamePostProcessor.java
package io.pageobject.generator.name;

import org.jsoup.nodes.Element;

import static io.pageobject.generator.element.Elements.isLink;

public class LinkPostfixNamePostProcessor extends PostfixNamePostProcessor {

    public LinkPostfixNamePostProcessor() {
        super("Link", new String[] {"Link", "Button"});
    }

    @Override
    public boolean isMatchingElement(Element element) {
        return isLink(element);
    }
}

File: lib/name/link-postfix-name-post-processor.js
"use strict";

const PostfixNamePostProcessor = require('./postfix-name-post-processor');
const isLink = require('../element/elements').isLink;

class LinkPostfixNamePostProcessor extends PostfixNamePostProcessor {

  constructor() {
    super('Link', ['Link', 'Button']);
  }

  isMatchingElement(element) {
    return isLink(element);
  }

}

module.exports = LinkPostfixNamePostProcessor;